SofaGLFW/src/SofaGLFW/SofaGLFWBaseGUI.h

```diff
@@ -138,11 +142,16 @@ class SOFAGLFW_API SofaGLFWBaseGUI
     int m_lastWindowPositionY{ 0 };
     int m_lastWindowWidth{ 0 };
     int m_lastWindowHeight{ 0 };
-
-    bool m_isMouseInteractionEnabled { false };
-    
-    std::shared_ptr<sofaglfw::BaseGUIEngine> m_guiEngine;
-
+    fixed_array<double, 16>  m_lastProjectionMatrix;
+    fixed_array<double, 16>  m_lastModelviewMatrix;
```

Comment:
Well even though the matrices are stored in the vparams, accessing them requires passing through the  setter and getter methods.


    void setModelViewMatrix( const double m[16] ) { for(unsigned i=0; i<16; i++) m_modelViewMatrix[i] = SReal(m[i]); }
    void getModelViewMatrix( double m[16] ) const { for(unsigned i=0; i<16; i++) m[i] = double(m_modelViewMatrix[i]); }

The members m_modelViewMatrix and m_projectionMatrix are protected, so they are not directly accessible outside the class.

In my implementation, I store these matrices in my own member variables to avoid repeatedly calling the getter methods, which would add unnecessary overhead. This allows me to access and use the matrices directly. Additionally, accessing the data of the matrices requires passing a matrix, so having a matrix member is necessary in any case.

Furthermore, this approach aligns with the existing pattern used in the drawScene method in QtGLViewer.cpp, where the matrices are similarly stored and managed within the class.